camel-idea-plugin/src/main/java/org/apache/camel/idea/service/extension/camel/JavaCamelIdeaUtils.java

```diff
@@ -92,7 +101,39 @@ public boolean isCameSimpleExpressionUsedAsPredicate(PsiElement element) {
                     method = ((PsiMethodCallExpression) exp).resolveMethod();
                     if (method != null) {
                         String name = method.getName();
-                        return Arrays.stream(SIMPLE_PREDICATE).anyMatch(name::equals);
+                        return Arrays.stream(PREDICATE_EIPS).anyMatch(name::equals);
+                    }
+                }
+            }
+            return false;
+        }
+        return false;
+    }
+
+    @Override
+    public boolean isCameJSonPathExpressionUsedAsPredicate(PsiElement element) {
+        // java
+        PsiMethodCallExpression call = PsiTreeUtil.getParentOfType(element, PsiMethodCallExpression.class);
+        if (call != null) {
+            // okay dive into the psi and find out which EIP are using the simple
+            PsiElement child = call.getFirstChild();
+            if (child instanceof PsiReferenceExpression) {
+                PsiExpression exp = ((PsiReferenceExpression) child).getQualifierExpression();
+                if (exp == null) {
+                    // okay it was not a direct method call, so see if it was passed in as a parameter instead (expression list)
+                    element = element.getParent();
+                    if (element instanceof PsiExpressionList) {
```

Comment:
Since it's not an else if condition, I guess the goal is to get the PsiMethodCallExpression.class ?

If that's the case it should be fine you just use `exp  = PsiTreeUtil.getParentOfType(element,PsiMethodCallExpression.class)`, since it will travers up in the tree until it find a parent instance of PsiMethodCallExpression.class.

Reply:
Maybe but its a copy of the existing working code we have for the simple language, and this should be identical, so I wont risk changing it.

Reply:
If it's a copy of existing code we should move it out to a common class for sharing

Reply:
If it's a copy of existing code we should move it out to a common class for sharing